instrumentation/opentelemetry-instrumentation-urllib3/tests/test_urllib3_integration.py

```diff
@@ -279,3 +279,28 @@ def response_hook(span, request, response):
         self.assertEqual(span.name, "name set from hook")
         self.assertIn("response_hook_attr", span.attributes)
         self.assertEqual(span.attributes["response_hook_attr"], "value")
+
+    def test_extended_request_hook(self):
+        def extended_request_hook(span, request, headers, body):
+            span.set_attribute("request_hook_headers", json.dumps(headers))
+            span.set_attribute("request_hook_body", body)
+
+        URLLib3Instrumentor().uninstrument()
+        URLLib3Instrumentor().instrument(
+            request_hook=extended_request_hook,
+        )
+
+        headers = {"header1": "value1", "header2": "value2"}
+        body = "param1=1&param2=2"
+
+        pool = urllib3.HTTPConnectionPool("httpbin.org")
+        response = pool.request("GET", "/status/200", body=body, headers=headers)
```

Comment:
Small comment - I'd change GET to POST so that it makes sense for the request to have a body